### PR TITLE
api: added pagination/ordering to repositories

### PR DESCRIPTION
**Pull request by vitoravelino**
Signed-off-by: Vítor Avelino <vavelino@suse.com>

Depends-on #2030

### CODE REVIEW

File: lib/api/helpers/ordering.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module API
+  module Helpers
+    module Ordering
```
Comment (mssola) [inline lib/api/helpers/ordering.rb]: Could you add documentation? Even if it's a bit obvious, I believe it's good practice to write what it's expected to happen when you include this module, or you use this method (e.g. the parameters that are going to be used)

File: lib/api/helpers/pagination.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+# Extracted and adapted from https://gitlab.com/gitlab-org/gitlab-ce/blob/master/lib/api/helpers/pagination.rb
+
+module API
+  module Helpers
+    module Pagination
```
Comment (mssola) [inline lib/api/helpers/pagination.rb]: Same here

File: lib/api/helpers/pagination.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      def paginate(relation)
+        return relation if params[:all]
```
Comment (mssola) [inline lib/api/helpers/pagination.rb]: Maybe it's safer:

```ruby
return relation if params[:all].present?
```

File: lib/api/helpers/ordering.rb (hunk 1 above)
Comment (vitoravelino) [reply lib/api/helpers/ordering.rb]: Sure.

File: lib/api/helpers/pagination.rb (hunk 3 above)
Comment (mssola) [reply lib/api/helpers/pagination.rb]: Just checked, it is not :sweat_smile: Ignore this.

File: lib/api/helpers/ordering.rb (hunk 1 above)
Comment (vitoravelino) [reply lib/api/helpers/ordering.rb]: Done.

**Pull request merged**